microsetta_admin/server.py

```diff
@@ -774,6 +775,191 @@ def scan():
                                       recorded_type)
 
 
+def get_legends(criteria):
+
+    dict = {}
+    hex = 'ABCDEF0123456789'
+    if criteria not in session:
+        status, fields = APIRequest.get(
+            '/api/admin/barcode_query_fields')
+
+        if status == 200:
+            for obj in fields:
+                if criteria == "Type" and obj["label"] == "Sample Site":
+                    for val in obj["values"]:
+                        code = [random.choice(hex) for i in range(6)]
+                        color_code = ['#' + ''.join(code)]
+                        while color_code[0] in dict:
+                            code = [random.choice(hex) for i in range(6)]
+                            color_code = ['#' + ''.join(code)]
+                        dict[obj["values"][val]] = color_code[0]
+
+                if criteria == "Status" and obj["label"] == "Sample Status":
+                    for val in obj["values"]:
+                        code = [random.choice(hex) for i in range(6)]
+                        color_code = ['#' + ''.join(code)]
+                        while color_code[0] in dict:
+                            code = [random.choice(hex) for i in range(6)]
+                            color_code = ['#' + ''.join(code)]
+                        tmp = obj["values"][val].replace(' ', '-').lower()
+                        dict[tmp] = color_code[0]
+
+                if criteria == "Project" and obj["label"] == "Project":
+                    for val in obj["values"]:
+                        if not ('-' in obj["values"][val]):
+                            code = [random.choice(hex) for i in range(6)]
+                            color_code = ['#' + ''.join(code)]
+                            while color_code[0] in dict:
+                                code = [random.choice(hex) for i in range(6)]
+                                color_code = ['#' + ''.join(code)]
+                            dict[obj["values"][val]] = color_code[0]
+
+            if len(dict) > 0:
+                dict['None'] = "#000000"
+                session[criteria] = dict
```

Comment:
During Visualization stage, samples cabe visualised on basis of three different criterion, i.e, project, sample status and types. For any critira, legentds and their respective colors are generated randomly and stored in a dict,. The purpose ofusing sessions is to preserve the generated dictionaries during a session. This will avoid generation of random color mappings for same criterio for visualization within single session if accessed multiple times.
For example if a person makes five requests for sample visualization on sample status during a session, the color map will not change and this will avoid confusion